src/VirtoCommerce.ExperienceApiModule.Core/Helpers/GraphTypeExtenstionHelper.cs

```diff
@@ -13,14 +13,49 @@ public static class GraphTypeExtenstionHelper
         public static Type GetActualType<TGraphType>() where TGraphType : IGraphType
         {
             var graphType = typeof(TGraphType);
+
+            return GetActualType(graphType);
+        }
+
+        public static Type GetActualType(Type graphType)
+        {
             var result = AbstractTypeFactory<IGraphType>.FindTypeInfoByName(graphType.Name)?.Type;
+
             if (result == null)
             {
                 result = graphType;
             }
+
             return result;
         }
 
+        /// For generic graph type definitions like NonNullGraphType ProdcutType
+        /// or NonNullGraphType ListGraphType ProdcutType
+        public static Type GetComplexType<TGraphType>() where TGraphType : IGraphType
```

Comment:
Can we rename this method to `GetActualComplexType`  to make its responsibility more clear?